src/store/initial_state.js

```diff
@@ -89,6 +89,10 @@ const state: GlobalState = {
             recent: {},
         },
         typing: {},
+        roles: {
+            roles: [],
```

Comment:
This should be an object, not an array

Reply:
You are right.

Reply:
Ah, good catch, this explains a really minor but odd bug I was seeing on the spinmint!

Reply:
Fixed